src/components/TimeSelectInput/index.js

```diff
@@ -0,0 +1,505 @@
+        this.setState({
+            inputFocusedIndex: 1,
+            minutes: '',
+        });
+    }
+
+    handleAmPmChange(value) {
+        this.handleChangeTime({
+            inputFocusedIndex: -1,
+            ampm: value,
+        });
+    }
+
+    hanldeFocusAmPm() {
+        this.setState({ inputFocusedIndex: 2 });
+    }
+
+    handleKeyDown(event) {
+        const { keyCode } = event;
+
+        if (keyCode === RIGHT_KEY) {
+            this.handleRightKeyPressed();
+        }
+        if (keyCode === LEFT_KEY) {
+            this.handleLeftKeyPressed();
+        }
+        if (keyCode === UP_KEY) {
+            this.incrementHandler();
+        }
+        if (keyCode === DOWN_KEY) {
+            this.decrementHandler();
+        }
+        if (keyCode === DELETE_KEY) {
+            this.resetState();
+        }
+    }
+
+    handleRightKeyPressed() {
+        const { inputFocusedIndex } = this.state;
+        const nextInputIndex = inputFocusedIndex + 1;
+        const nextInputToFocus = this.inputsMap[nextInputIndex];
+        if (nextInputToFocus && nextInputToFocus.current != null) {
+            this.setState({ inputFocusedIndex: inputFocusedIndex + 1 });
+            nextInputToFocus.current.focus();
+        }
+    }
+
+    handleLeftKeyPressed() {
+        const { inputFocusedIndex } = this.state;
+        const prevInputIndex = inputFocusedIndex - 1;
+        const prevInputToFocus = this.inputsMap[prevInputIndex];
+        if (prevInputToFocus) {
+            this.setState({ inputFocusedIndex: inputFocusedIndex - 1 });
+            prevInputToFocus.current.focus();
+        }
+    }
+
+    incrementHandler() {
+        const { inputFocusedIndex } = this.state;
+        this.isUpOrDownKeyPressed = true;
+        if (inputFocusedIndex === 0) {
+            this.incrementHour();
+        }
+        if (inputFocusedIndex === 1) {
+            this.incrementMinutes();
+        }
+        if (inputFocusedIndex === 2) {
+            event.preventDefault();
+            this.setNextAmPmValue();
+        }
+    }
+
+    decrementHandler() {
+        const { inputFocusedIndex } = this.state;
+        this.isUpOrDownKeyPressed = true;
+        if (inputFocusedIndex === 0) {
+            this.decrementHour();
+        }
+        if (inputFocusedIndex === 1) {
+            this.decrementMinutes();
+        }
+        if (inputFocusedIndex === 2) {
+            event.preventDefault();
+            this.setNextAmPmValue();
+        }
+    }
+
+    handleButtonsDown() {
+        this.isUpOrDownButtonPressed = true;
+    }
+
+    handleButtonsFocus() {
+        const { inputFocusedIndex } = this.state;
+        const currentInputFocused = this.inputsMap[inputFocusedIndex];
+        if (currentInputFocused) {
+            currentInputFocused.current.focus();
+        }
+    }
+
+    resetState() {
+        const { inputFocusedIndex } = this.state;
+        if (inputFocusedIndex === 0) {
+            this.handleChangeTime()({
```

Comment:
sure this is the problem resetting the hour value you are calling it twice and the first time without arguments:
```this.handleChangeTime()({ ... })```